File: audio.R

setwd("D:\\nonstruct")
library(tuneR)


# 读音频文件
x <- readMP3("beibei.mp3")
x

# 绘图
plot(x)


# 写音频文件
writeWave(x, "beibei.wav")


# 短时傅里叶变换
library(seewave) 
d1 <- spectro(x, wl = 512)


# 提取LFCC
m1 <- melfcc(x)
dim(m1)

image(m1)

File: audio.py

import os
os.chdir("D:\\nonstruct")


# 读音频文件
import librosa
x, sr = librosa.load("beibei.mp3", sr = 44100, mono = False)
x.shape
sr


# 绘图
import matplotlib.pyplot as plt
import librosa.display
librosa.display.waveplot(x, sr = sr)
plt.show()


# 写音频文件
librosa.output.write_wav("beibei.wav", x, sr)


# 短时傅里叶变换
d1 = librosa.stft(x, n_fft = 512)


# 提取LFCC
m1 = librosa.feature.mfcc(x, sr, n_mfcc=12)
m1.shape

librosa.display.specshow(m1, sr = sr, x_axis = "time")
plt.show()












